src/Npgsql/PostgresEnvironment.cs

```diff
@@ -22,6 +23,43 @@ static class PostgresEnvironment
 
         public static string? TimeZone => Environment.GetEnvironmentVariable("PGTZ");
 
+        public static string? Options => Environment.GetEnvironmentVariable("PGOPTIONS");
+
+        public static Dictionary<string, string> ParsedOptions
+        {
+            get
+            {
+                if (_optionsCache == Options)
```

Comment:
Very small nit: if you inverse the logic and the parsing and assignment inside the if block, you get rid of one of the two `return ParseOptionsCache`

Reply:
OK. I actually inverted it because it saves me a little indentation but I agree that returning early should probably be reserved for more complex cases.